### PR TITLE
Added a first attempt at reweighted SNR for GRB search

### PR DESCRIPTION
**Pull request by idorrington92**

### CODE REVIEW

File: bin/pycbc_multi_inspiral [hunk 1]
```diff
@@ -142,6 +142,39 @@ logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
 inj_filter_rejector = pycbc.inject.InjFilterRejector.from_cli_multi_ifos(opt,opt.instruments)
 ctx = scheme.from_cli(opt)
 
+def network_chisq(chisq, chisq_dof, snr_dict):
```
Comment (spxiwh) [inline bin/pycbc_multi_inspiral]: Check that the chisq-dof here is *actually* the number of frequency bins, and not the actual DOF.

File: bin/pycbc_multi_inspiral [hunk 2]
```diff
@@ -142,6 +142,39 @@ logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
...
+    ifos = sorted(snr_dict.keys())
+    chisq_per_dof = {ifo : chisq[ifo] / (2*chisq_dof[ifo] - 2) for ifo in ifos}
+    snr2 = {ifo : np.array(snr_dict[ifo]) *
+                np.array(snr_dict[ifo]).conj() for ifo in ifos}
+    coinc_snr2 = sum(snr2.values())
+    snr2_ratio = {ifo : snr2[ifo] / coinc_snr2 for ifo in ifos}
+    network_chisq = sum( np.array(chisq_per_dof.values()) *
+                    np.array(snr2_ratio.values()) )
```
Comment (spxiwh) [inline bin/pycbc_multi_inspiral]: While `values()` should give things ordered in the same way, I wouldn't assume this. Explicitly loop over ifos (or do a list comprehension) and select chisq_per_dof[ifo] * snr2_ratio[ifo]

File: bin/pycbc_multi_inspiral [hunk 3]
```diff
@@ -142,6 +142,39 @@ logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
 inj_filter_rejector = pycbc.inject.InjFilterRejector.from_cli_multi_ifos(opt,opt.instruments)
 ctx = scheme.from_cli(opt)
 
+def network_chisq(chisq, chisq_dof, snr_dict):
+    ifos = sorted(snr_dict.keys())
+    chisq_per_dof = {ifo : chisq[ifo] / (2*chisq_dof[ifo] - 2) for ifo in ifos}
+    snr2 = {ifo : np.array(snr_dict[ifo]) *
+                np.array(snr_dict[ifo]).conj() for ifo in ifos}
```
Comment (spxiwh) [inline bin/pycbc_multi_inspiral]: You might need to cast this to real.

**Comment by spxiwh** [comment]
A wider question is to consider whether you want to compute the `newSNR` like statistics and store them in the output file. The all-sky search does not do this, `newSNR` is just computed from the SNR and chisq values if needed. I guess it's a tradeoff between filesize and, perhaps, speed.

For testing, while the PP code is being rewritten this is surely helpful, but maybe consider this before it gets hardcoded into the post-processing.

File: bin/pycbc_multi_inspiral (hunk 1 above)
Comment (idorrington92) [reply bin/pycbc_multi_inspiral]: Looking through the singleDetPowerChisq code here, I think it actually is the DOF that is output. Changing the code now.

**Comment by idorrington92** [comment]
Don't merge this. I'm debugging.

**Comment by idorrington92** [comment]
I think I've finished debugging it now.

**Pull request merged**